REGRESSION(267415@main): Broke python autoinstaller for python 3.11

mcatanzaro opened the pull request:
#### 160f1a456bf42fc2e2e108ed207be765a8a68d7d
<pre>
REGRESSION(267415@main): Broke python autoinstaller for python 3.11
<a href="https://bugs.webkit.org/show_bug.cgi?id=260889">https://bugs.webkit.org/show_bug.cgi?id=260889</a>

Reviewed by Jonathan Bedard.

All of these python libraries are quite interdependent. Apparently
beautifulsoup4 depends on functionality removed from the newer
setuptools.

* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:

Canonical link: <a href="https://commits.webkit.org/267461@main">https://commits.webkit.org/267461@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/03789890fb275c46c833b6492ec96027fd63a9a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16569 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16892 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17334 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18351 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15539 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16759 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20122 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17033 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17871 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16767 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17169 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14334 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19133 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/16694 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14413 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15018 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21803 "19 flakes 80 failures") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15401 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15184 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19493 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15782 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13400 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/16451 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14980 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3989 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19349 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15608 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
```diff
@@ -58,7 +58,7 @@
         AutoInstall.register(Package('win_inet_pton', Version(1, 1, 0), pypi_name='win-inet-pton'))
 
-if sys.version_info >= (3, 8):
+if sys.version_info >= (3, 12):
```
Hmm, so this keeps BeautifulSoup broken on Python 3.12.

https://github.com/pypa/setuptools/pull/2731, shipped in `setuptools==58`, seems to be what dropped support for `2to3` (which the old Python 2-supporting BeautifulSoup relies on… again pain from us trying to support Python 2).

https://github.com/pypa/setuptools/pull/3685 I think is the last fix for `setuptools` on Python 3.12 (shipped in `setuptools==66.1.0`).

I suppose this is unsurprising, given the underlying reason for the breakage is the deprecation of 2to3 (whose removal keeps getting pushed back, but is currently gone in 3.13, but was originally expected to be gone by now).

Given we still need to fix the rest of the versions for Python 3.12, I suppose this is fine to land for now, so a non-reviewer r+ from me.

mcatanzaro replied:
The issue with BeautifulSoup (whatever that is) should be fixed by #17201, hopefully.

mcatanzaro committed:
REGRESSION(267415@main): Broke python autoinstaller for python 3.11
https://bugs.webkit.org/show_bug.cgi?id=260889

Reviewed by Jonathan Bedard.

All of these python libraries are quite interdependent. Apparently
beautifulsoup4 depends on functionality removed from the newer
setuptools.

* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:

Canonical link: https://commits.webkit.org/267461@main
